File: beginner_tutorials/src/add_two_ints_action_client.cpp
#include <beginner_tutorials/AddTwoIntsAction.h>
#include <actionlib/client/simple_action_client.h>

typedef actionlib::SimpleActionClient<beginner_tutorials::AddTwoIntsAction> Client;

int main(int argc, char** argv)
{
    ros::init(argc, argv, "add_two_ints_client");
    Client client("add_two_ints", true); // true -> don't need ros::spin()
    client.waitForServer();
    beginner_tutorials::AddTwoIntsGoal goal;
    goal.a = 10;
    goal.b = 20;
    // Fill in goal here
    client.sendGoal(goal);
    client.waitForResult(ros::Duration(5.0));
    if (client.getState() == actionlib::SimpleClientGoalState::SUCCEEDED)
      ROS_INFO("add two int returns %ld", client.getResult()->sum);
    printf("Current State: %s\n", client.getState().toString().c_str());
    return 0;
}

File: beginner_tutorials/scripts/add_two_ints_action_client.py
#! /usr/bin/env python

import rospy

import actionlib

import beginner_tutorials.msg

def add_two_ints_client():
    client = actionlib.SimpleActionClient('add_two_ints', beginner_tutorials.msg.AddTwoIntsAction)
    client.wait_for_server()
    goal = beginner_tutorials.msg.AddTwoIntsGoal(a=20, b=30)
    client.send_goal(goal)
    client.wait_for_result()

    return client.get_result()  # A AddTwoIntsResult

if __name__ == '__main__':
    try:
        rospy.init_node('add_two_ints_client_py')
        result = add_two_ints_client()
        rospy.loginfo("Result: {}".format(result))
    except rospy.ROSInterruptException:
        rospy.logerr("program interrupted before completion")
